src/utils.ts

```diff
@@ -2177,67 +2251,115 @@ export class newAxis {
-    const orientation = this.isVertical ? -90 : 0;
     const textParams: textParams = {
-      width: this.drawLength, fontsize: this.FONT_SIZE, font: this.FONT, align: 'center',
-      baseline: baseline, style: 'bold', orientation: orientation
+      width: this.drawLength, fontsize: this.FONT_SIZE, font: this.FONT, align: align, color: this.strokeStyle,
+      baseline: baseline, style: 'bold', orientation: orientation, backgroundColor: "hsla(0, 0%, 100%, 0.5)"
     };
     const textName = new newText(this.title, nameCoords, textParams);
     textName.draw(context);
   }
 
+  private topArrowTitleProperties(): [Vertex, string, string, number] {
+    let nameCoords = this.end.copy();
+    let alignChoices = ["start", "end"];
+    let signFontAdd = 1;
+    if (!this.isVertical) {
+      alignChoices = ["end", "start"];
+      signFontAdd = -0.8;
+      nameCoords.y += this.FONT_SIZE;
+    }
+    nameCoords.x += signFontAdd * this.FONT_SIZE;
+    return [nameCoords, alignChoices[this.verticalPickIdx()], "middle", 0]
+  }
+
+  private centeredTitleProperties(): [Vertex, string, string, number] {
+    let baseline = ['bottom', 'top'][this.horizontalPickIdx()];
+    let nameCoords = this.end.add(this.origin).divide(2);
+    if (this.isVertical) {
+      nameCoords.x -= this.offsetTitle;
+      baseline = ['bottom', 'top'][this.verticalPickIdx()];
+    } else {
+      nameCoords.y -= this.offsetTitle;
+    }
+    return [nameCoords, "center", baseline, this.isVertical ? -90 : 0]
+  }
+
   private drawTicks(context: CanvasRenderingContext2D, pointHTMatrix: DOMMatrix, color: string) {
     const ticksCoords = [];
+    const ticksText: newText[] = [];
+    const tickTextParams = this.computeTickTextParams();
+
     let count = Math.max(0, this.ticks[0]);
     this.ticks.forEach((tick, idx) => {
       if (tick >= this.minValue && tick <= this.maxValue) {
+        context.setTransform(pointHTMatrix);
         let point = this.drawTickPoint(context, tick, this.isVertical, pointHTMatrix, color);
+        let text = this.labels[idx];
         ticksCoords.push(point);
-        let text = this.labels[idx]
         if (this.isDiscrete) {
           if (count == tick && this.labels[count]) { text = this.labels[count]; count++ }
           else { text = '' }
         }
-        this.drawTickText(context, text, point, pointHTMatrix);
+        ticksText.push(this.computeTickText(context, text, tickTextParams, point, pointHTMatrix));
       }
     })
+    this.ticksFontsize = Math.min(...ticksText.map(tickText => tickText.fontsize));
+    context.resetTransform();
+    ticksText.forEach(tickText => {
+      tickText.fontsize = this.ticksFontsize;
+      tickText.width = null;
+      tickText.draw(context);
+    })
     return ticksCoords
   }
 
+  private computeTickTextParams(): textParams {
+    const [textAlign, baseline] = this.textAlignments();
+    let textWidth = null;
+    let textHeight = null;
+    if (['start', 'end'.includes(textAlign)]) {
```

Comment:
Possible typo here. Don't you want to write `['start', 'end'].includes(textAlign)` instead ?

Reply:
Thanks